### PR TITLE
Feature/first

### PR DESCRIPTION
**Pull request by ezstarr**
i think this feature is ready to be added into production

### CODE REVIEW

**Comment by ezstarr** [comment]
cherry picked the feature/point code out of this branch.

**Review by ezstarr — COMMENTED** [review]
Took out more overlooked points code

File: SCHEMA.sql [hunk 1]
```diff
@@ -1,11 +1,11 @@
--- Active: 1699682544906@@localhost@5432@timeenjoyed
+-- Active: 1703653682760@@127.0.0.1@5432@timebot_test
 CREATE TABLE IF NOT EXISTS users (
     uid BIGINT PRIMARY KEY,
     discord_id BIGINT UNIQUE,
     twitch_id BIGINT UNIQUE,
     moderator BOOLEAN NOT NULL DEFAULT false,
     token TEXT UNIQUE NOT NULL,
-    created TIMESTAMP DEFAULT (now() at time zone 'utc')
+    created TIMESTAMP DEFAULT (now() at time zone 'utc'),
```
Comment (EvieePy) [inline SCHEMA.sql]: This will break the SQL (The comma at the end)

File: extensions/twitch/general.py [hunk 2]
```diff
@@ -184,6 +258,12 @@ async def midnight(self) -> None:
 
                 await channel.send(f"it's midnight, the {day_str} in {timezone}!")
 
+    @commands.command()
```
Comment (EvieePy) [inline extensions/twitch/general.py]: This shouldn't be in production

File: routes/eventsub.py [hunk 3]
```diff
@@ -132,7 +133,9 @@ async def notifcation_event(self, data: dict[str, Any], /) -> None:
             await self.online_event(event["broadcaster_user_login"], event["broadcaster_user_id"])
 
         elif type_ == "channel.channel_points_custom_reward_redemption.add":
+            # step 1 of first_redeem
             logger.info("EventSub Redemption received.")
+            print(f"================================the event id is: {event_id}")
```
Comment (EvieePy) [inline routes/eventsub.py]: This shouldn't be in production

File: routes/first.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+    # this path is redeems/first
+    @route("/first", methods=["GET"])  # (2/3 component - displays the html page)
+    async def get_first_streak(self, request: Request) -> Response:
+        """this loads the index.html file at this path."""
+
+        # return HTMLResponse(thing)
+        return FileResponse("web/first-redeem/index.html")
+
+    @route("/random/data", methods=["GET"])
```
Comment (EvieePy) [inline routes/first.py]: This doesn't need to exist anymore?

File: routes/first.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+        html: str = f"""
+        <div>
+            Radnom number: {num}!
+        </div>
+        """
+
+        return HTMLResponse(html)
+
+    @route("/random", methods=["GET"])
```
Comment (EvieePy) [inline routes/first.py]: This also doesn't need to exist?

File: routes/sse.py [hunk 6]
```diff
@@ -88,3 +88,61 @@ async def publisher() -> AsyncGenerator[dict[str, Any], Any]:
...
+
+            except asyncio.CancelledError as e:
+                logger.info('EventSource "%s" connection closed: %s', id_, e)
+
+                del self.app.htmx_listeners[id_]
+                raise e
+
+        return EventSourceResponse(publisher())
+
+    @route("/random", methods=["GET"])
```
Comment (EvieePy) [inline routes/sse.py]: This doesn't need to exist?

**Review by EvieePy — CHANGES_REQUESTED** [review]

File: routes/first.py (hunk 4 above)
Comment (ezstarr) [reply routes/first.py]: shame, such pretty example code though xD

**Review by EvieePy — APPROVED** [review]

**Pull request merged**